### DISCUSSION TITLE
The x-axis moves in only one direction.

### DISCUSSION THREAD

**Discussion by kotovasia5120** [kachurovskiy/nanoels#224] [Q&A] [answered]
The x-axis moves in only one direction.
I tested the motor with a simple sketch that generated pulses and rotated in both directions without problems. With the firmware it only moves in one direction.

**Comment by kachurovskiy** [comment]
@kotovasia5120 did you modify the firmware in some way? Please post your code at https://gist.github.com/ and share the link here.

Try testing the voltage on the driver DIR terminal when going either direction. It might not be clearing the threshold due to other terminals being in use and lowering the bus voltage.

**Reply by kotovasia5120** [reply]
https://gist.github.com/kotovasia5120/57a8aa33c82d2aaf625c2873dc2c87df
With the sketch below the motor rotates in both directions. Later I will check the signal voltage again.
https://gist.github.com/kotovasia5120/d2f1dc46bc60d46eecae3d5582635b98

**Reply by kachurovskiy** [reply]
Looks like you also changed the DIR pin mode?

![image](https://github.com/user-attachments/assets/ab0d3ce8-295b-42f4-a0ae-f8f009952b8d)

I think that change needs to be rolled back - https://chatgpt.com/share/67272eb5-9f84-8002-b7bd-4a8608260c11

**Reply by kotovasia5120** [reply]
The common anode is connected to 3.3v OUTPUT_OPEN_DRAIN pulls signals to ground.

**Reply by kotovasia5120** [reply]
The z axis works as expected.

**Reply by kotovasia5120** [reply]
@kachurovskiy 
It's the same with pinMODE OUTPUT, here are the signals from the outputs.
DIR Working direction
![001](https://github.com/user-attachments/assets/99baba3e-591f-47f3-8213-2ccab96cf4bd)
STEP Working direction
![003](https://github.com/user-attachments/assets/c4dda8e3-4cac-4bc1-9fe4-64fda6af005c)
DIR NotWorking direction
![002](https://github.com/user-attachments/assets/ebc35558-296d-4426-9906-f94b3f03699b)
STEP Working direction
![000](https://github.com/user-attachments/assets/82ad72c0-5bae-42bb-ba6e-95e44e265de9)

**Comment by kotovasia5120** [comment, marked as answer]
I came to the conclusion that the outputs were out of order.
